x-pack/plugins/security_solution/public/resolver/view/panels/panel_content_related_detail.tsx

```diff
@@ -0,0 +1,340 @@
+      return event.eventType(evt) === specificCategory ? sumtotal + 1 : sumtotal;
+    }, 0);
+    return [specificEvent, countOfCategory, specificCategory || naString];
+  }, [relatedEventsForThisProcess, naString, relatedEventId]);
+
+  const [sections, formattedDate] = useMemo(() => {
+    if (!relatedEventToShowDetailsFor) {
+      // This could happen if user relaods from URL param and requests an eventId that no longer exists
+      return [[], naString];
+    }
+    // Assuming these details (agent, ecs, process) aren't as helpful, can revisit
+    const {
+      agent,
+      ecs,
+      process,
+      ...relevantData
+    } = relatedEventToShowDetailsFor as ResolverEvent & {
+      ecs: unknown;
+    };
+    let displayDate = '';
+    const sectionData: Array<{
+      sectionTitle: string;
+      entries: Array<{ title: string; description: string }>;
+    }> = Object.entries(relevantData)
+      .map(([sectionTitle, val]) => {
+        if (sectionTitle === '@timestamp') {
+          displayDate = formatDate(val);
+          return { sectionTitle: '', entries: [] };
+        }
+        if (typeof val !== 'object') {
+          return { sectionTitle, entries: [{ title: sectionTitle, description: `${val}` }] };
+        }
+        return { sectionTitle, entries: [...objectToDescriptionListEntries(val)] };
+      })
+      .filter((v) => v.sectionTitle !== '' && v.entries.length);
+    return [sectionData, displayDate];
+  }, [relatedEventToShowDetailsFor, naString]);
+
+  const waitCrumbs = useMemo(() => {
+    return [
+      {
+        text: eventsString,
+        onClick: () => {
+          pushToQueryParams({ crumbId: '', crumbEvent: '' });
+        },
+      },
+    ];
+  }, [pushToQueryParams, eventsString]);
+
+  const crumbs = useMemo(() => {
+    return [
+      {
+        text: eventsString,
+        onClick: () => {
+          pushToQueryParams({ crumbId: '', crumbEvent: '' });
+        },
+      },
+      {
+        text: processName,
+        onClick: () => {
+          pushToQueryParams({ crumbId: processEntityId!, crumbEvent: '' });
+        },
+      },
+      {
+        text: (
+          <>
+            <EuiI18nNumber value={totalCount} />
+            {/* Non-breaking space->*/ ` ${eventsString}`}
+          </>
+        ),
+        onClick: () => {
+          pushToQueryParams({ crumbId: processEntityId!, crumbEvent: 'all' });
+        },
+      },
+      {
+        text: (
+          <>
+            <EuiI18nNumber value={countBySameCategory} />
+            {/* Non-breaking space->*/ ` ${relatedEventCategory}`}
+          </>
+        ),
+        onClick: () => {
+          pushToQueryParams({
+            crumbId: processEntityId!,
+            crumbEvent: relatedEventCategory || 'all',
+          });
+        },
+      },
+      {
+        text: relatedEventToShowDetailsFor
+          ? event.descriptiveName(relatedEventToShowDetailsFor)

```

Comment:
`event.descriptiveName` only takes the first `event.category` field. I haven't looked through all the code yet, but let's say `event.category: [process, file]`.

In the related event drop down we'll show:
```
1 Process
1 File
```

If a user clicks on `1 File` will `event.descriptiveName` show `process`?

Sorry if I'm confusing how the flow works too.

Reply:
I'd like to discuss this on the call today. What makes it a little more complicated is that we flex the category into the query param as the "current category".

In _this_ particular case, it does _nothing_ when you click on it, because it's the last crumb (just there as an indication of "where you are" in nav. not a link to anything) but your overall point is valid and relevant in a few other places.